File: backend/src/main/java/com/dongnemarket/auction/dto/AuctionResponse.java
package com.dongnemarket.auction.dto;

import com.dongnemarket.auction.entity.Auction;
import com.dongnemarket.auction.entity.AuctionStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public class AuctionResponse {

    private final Long auctionId;
    private final Long sellerId;
    private final String title;
    private final String imageUrl;
    private final String description;
    private final BigDecimal currentPrice;
    private final Long highestBidderId;
    private final AuctionStatus status;
    private final LocalDateTime endAt;
    private final LocalDateTime createdAt;

    private AuctionResponse(Long auctionId, Long sellerId, String title, String imageUrl, String description,
                            BigDecimal currentPrice, Long highestBidderId, AuctionStatus status,
                            LocalDateTime endAt, LocalDateTime createdAt) {
        this.auctionId = auctionId;
        this.sellerId = sellerId;
        this.title = title;
        this.imageUrl = imageUrl;
        this.description = description;
        this.currentPrice = currentPrice;
        this.highestBidderId = highestBidderId;
        this.status = status;
        this.endAt = endAt;
        this.createdAt = createdAt;
    }

    public static AuctionResponse from(Auction a) {
        return new AuctionResponse(
                a.getId(), a.getSellerId(), a.getTitle(), a.getImageUrl(), a.getDescription(),
                a.getCurrentPrice(), a.getHighestBidderId(), a.getStatus(),
                a.getEndAt(), a.getCreatedAt());
    }

    public Long getAuctionId() { return auctionId; }
    public Long getSellerId() { return sellerId; }
    public String getTitle() { return title; }
    public String getImageUrl() { return imageUrl; }
    public String getDescription() { return description; }
    public BigDecimal getCurrentPrice() { return currentPrice; }
    public Long getHighestBidderId() { return highestBidderId; }
    public AuctionStatus getStatus() { return status; }
    public LocalDateTime getEndAt() { return endAt; }
    public LocalDateTime getCreatedAt() { return createdAt; }
}

File: backend/src/main/kotlin/com/dongnemarket/auction/dto/AuctionResponse.kt
package com.dongnemarket.auction.dto

import com.dongnemarket.auction.entity.Auction
import com.dongnemarket.auction.entity.AuctionStatus
import java.math.BigDecimal
import java.time.LocalDateTime

/**
 * 경매 상세·목록 응답.
 *
 * `auctionId` 는 Auction.id 가, `createdAt` 은 BaseTimeEntity.createdAt 이 nullable 이라
 * 컴파일러가 nullable 을 강제한다(둘 다 저장 전에는 값이 없다).
 */
@ConsistentCopyVisibility
data class AuctionResponse private constructor(
    val auctionId: Long?,
    val sellerId: Long,
    val title: String,
    val imageUrl: String?,
    val description: String?,
    val currentPrice: BigDecimal,
    val highestBidderId: Long?,
    val status: AuctionStatus,
    val endAt: LocalDateTime,
    val createdAt: LocalDateTime?,
) {
    companion object {
        @JvmStatic
        fun from(a: Auction): AuctionResponse =
            AuctionResponse(
                a.id,
                a.sellerId,
                a.title,
                a.imageUrl,
                a.description,
                a.currentPrice,
                a.highestBidderId,
                a.status,
                a.endAt,
                a.createdAt,
            )
    }
}
